<?xml version="1.0"?>
<launch>
    
    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lego_loam)/launch/localization.rviz" />

    <!--- TF -->
    <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 1.570795   0        1.570795 /map    /camera_init 10" />
    <node pkg="tf" type="static_transform_publisher" name="odom_to_base" args="0 0 0 -1.570795 -1.570795 0         /laser_odom /base_link   10" />
    <node pkg="tf" type="static_transform_publisher" name="base_to_velo" args="0 0 0 0 0 0        /base_link /velodyne   10" />



    <!-- LeGO-LOAM -->
    <node pkg="lego_loam" type="imageProjection"    name="imageProjection"    output="screen"/>
    <node pkg="lego_loam" type="featureAssociation" name="featureAssociation" output="screen"/>
    <node pkg="lego_loam" type="mapOptmization"     name="mapOptmization"     output="screen">
        <param name="save_map" value="false"/>
        <param name="model_path" value="c:\opt\ros\melodic\test_ws\src\global-LeGO-LOAM\train\locnet_descriptor510.pt"/>
    </node>
    <node pkg="lego_loam" type="transformFusion"    name="transformFusion"    output="screen"/>

    <!--- Initialize -->    
    <node pkg="lego_loam" type="globalLocalization"    name="globalLocalization"    output="screen">
        <param name="model_path" value="c:\opt\ros\melodic\test_ws\src\global-LeGO-LOAM\train\locnet_descriptor510.pt"/>
        <param name="feature_cloud_path" value="C:\Users\Haeyeon Kim\Desktop\lego_loam_result\feature_cloud.pcd"/>
        <param name="search_radius" value="1000.0"/>
    </node>

    <!--- Particle filter -->
    <node pkg="lego_loam" type="particleFilter"    name="particleFilter"    output="screen">
        <param name="num_particles" value="500"/>
        <param name="sampling_covariance" value="0.1"/>
        <param name="resample_var_x" value="0.1"/>
        <param name="resample_var_y" value="0.1"/>
        <param name="resample_var_yaw" value="0.1"/>
        <param name="map_downsample_x" value="20.0"/>
        <param name="map_downsample_x" value="20.0"/>
        <param name="map_downsample_x" value="20.0"/>
        
        <param name="rot_x" value="0.0"/>
        <param name="rot_y" value="-1.570795"/>
        <param name="rot_z" value="-1.570795"/>
    </node>
</launch>
